### PR TITLE
source-rootly contribution from gemsteam

### LINKED ISSUE #46042: [connector-request] Rootly
This is a **suggestion** to create a connector for **Rootly** using the UI Connector Builder.
The website for the connector is https://rootly.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by gemsteam**
## What
Closes https://github.com/airbytehq/airbyte/issues/46042
This PR adds a new connector for source Rootly (source-rootly).

Contributor-provided connector description:

Documentation: https://rootly.com/api#/
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| dashboards | id | DefaultPaginator | ✅ |  ❌  |
| incidents | id | DefaultPaginator | ✅ |  ❌  |
| authorizations | id | DefaultPaginator | ✅ |  ❌  |
| dashboards_panels | id | DefaultPaginator | ✅ |  ❌  |
| users | id | DefaultPaginator | ✅ |  ❌  |
| incident_types | id | DefaultPaginator | ✅ |  ❌  |
| ip_ranges | id | DefaultPaginator | ✅ |  ❌  |
| users_notification_rules | id | DefaultPaginator | ✅ |  ❌  |
| teams | id | DefaultPaginator | ✅ |  ❌  |
| sub_statuses | id | DefaultPaginator | ✅ |  ❌  |
| incidents_sub_statuses | id | DefaultPaginator | ✅ |  ❌  |
| incident_roles | id | DefaultPaginator | ✅ |  ❌  |
| post_mortems | id | DefaultPaginator | ✅ |  ❌  |
| status-pages | id | DefaultPaginator | ✅ |  ❌  |
| severities | id | DefaultPaginator | ✅ |  ❌  |
| services | id | DefaultPaginator | ✅ |  ❌  |
| incident_permission_sets | id | DefaultPaginator | ✅ |  ❌  |
| incident_permission_sets_resources | id | DefaultPaginator | ✅ |  ❌  |
| incident_permission_sets_booleans | id | DefaultPaginator | ✅ |  ❌  |
| roles | id | DefaultPaginator | ✅ |  ❌  |
| incidents_events | id | DefaultPaginator | ✅ |  ❌  |
| workflows | id | DefaultPaginator | ✅ |  ❌  |
| workflow_groups | id | DefaultPaginator | ✅ |  ❌  |
| workflows_tasks | id | DefaultPaginator | ✅ |  ❌  |
| form_sets | id | DefaultPaginator | ✅ |  ❌  |
| form_fields | id | DefaultPaginator | ✅ |  ❌  |
| playbooks | id | DefaultPaginator | ✅ |  ❌  |
| incidents_action_items | id | DefaultPaginator | ✅ |  ❌  |
| retrospective_configurations | id | DefaultPaginator | ✅ |  ❌  |
| retrospective_processes | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [x] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/rootly.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `api_key` | `string` | API Key.  |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| dashboards | id | DefaultPaginator | ✅ |  ❌  |
```
Comment (natikgadzhi) [inline docs/integrations/sources/rootly.md:13]: Wow, nice list of streams! 👏🏼

File: docs/integrations/sources/rootly.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Rootly
+Documentation: https://rootly.com/api#/
```
Comment (natikgadzhi) [inline docs/integrations/sources/rootly.md:2]:
```suggestion
Documentation: [https://rootly.com/api#/](https://rootly.com/api#/)
```

**Review by natikgadzhi — COMMENTED** [review]
Wow, this is very promising. Good work!

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@gemsteam can you sign the CLA please?

File: docs/integrations/sources/rootly.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `api_key` | `string` | API Key.  |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| dashboards | id | DefaultPaginator | ✅ |  ❌  |
+| incidents | id | DefaultPaginator | ✅ |  ❌  |
```
[the remark this answers is not in the document]
Comment (marcosmarxm) [reply docs/integrations/sources/rootly.md:14]: What about adding a transformation to move `updated_at` to higher level?
image

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Please check my latest comment about how implement incremental.

**Pull request closed without merging**